src/vs/workbench/services/workspaces/common/workspaceTrust.ts

```diff
@@ -126,23 +126,26 @@ export class WorkspaceTrustManagementService extends Disposable implements IWork
 	private registerListeners(): void {
 
 		// Resolve the workspace uris and resolve the initialization promise
-		this.resolveCanonicalWorkspaceUris().then(async () => {
-			this._initialized = true;
-			await this.updateWorkspaceTrust();
-
-			this._workspaceResolvedPromiseResolve();
-			if (!this.environmentService.remoteAuthority) {
-				this._workspaceTrustInitializedPromiseResolve();
-			}
-		});
+		this.resolveCanonicalWorkspaceUris()
+			.then(async () => {
+				this._initialized = true;
```

Comment:
is the intent that we stay uninitialized?

Reply:
This is a field that represents an internal state to the service. The promises, which are the signals to the outside world will be resolved. But if the canonical URIs could not be resolved, I think that it is safe to stay in restricted mode which we can ensure by not setting `this._initialized = true;`